doc/source/dev/missing-bits.rst

```diff
@@ -48,6 +48,46 @@ additional parameters can be added to the function anywhere after the
 ``*``; new parameters do not have to be added after the existing parameters.
 
 
+Return Objects
+~~~~~~~~~~~~~~
+For new functions or methods that return two or more conceptually distinct
+elements, return the elements in an object type that is not iterable. In
+particular, do not return a ``tuple``, ``namedtuple``, or a "bunch" produced
+by `~scipy._lib._bunch.make_tuple_bunch`, the latter being reserved for adding
+new attributes to iterables returned by existing functions. Instead, use an
+existing return class (e.g. `~scipy.optimize.OptimizeResult`), a new, custom
+return class, or use `~dataclasses.make_dataclass`.
+
+This practice of returning non-iterable objects forces callers to be more
+explicit about the element of the returned object that they wish to access,
+and it makes it easier to extend the function or method in a backward
+compatible way.
+
+If the return class is simple and not public (i.e. importable from a public
+module), it may be documented like::
+
+    Returns
+    -------
+    res : MyResultObject
+        An object with attributes:
+
+        attribute1 : ndarray
+            Customized description of attribute 1.
+        attribute2 : ndarray
+            Customized description of attribute 2.
+
+Here "MyResultObject" above does not link to external documentation because it
+is simple enough to fully document all attributes immediately below its name.
+
+Some return classes are sufficiently complex to deserve their own rendered
+documentation. This is fairly standard if the return class is public, but
+return classes should only be public if 1) they are intended to be imported by
+end-users and 2) if they have been approved by the mailing list. For complex,
```

Comment:
I would remove the approval part. To me this would be part of the same discussion for the new feature. Since it's for a new feature otherwise you would use bunch.

Reply:
Sure, it can be part of the same discussion, but I think it's worth mentioning somehow that the introduction of a new public class should be explicitly mentioned in addition to the feature itself in those emails that are sent to the mailing list. Can you suggest a wording that includes that idea, too?

Reply:
Resolving this. Indeed this does not say we have to send 2 emails, this just re state that we need mailing list approval for API changes which is our status quo 👍